cli/src/pcluster/utils.py

```diff
@@ -1247,3 +1247,25 @@ def get_instance_gpus(instance_type, instance_info=None):
     gpus = sum([gpus.get("Count") for gpus in gpu_info.get("Gpus")]) if gpu_info else 0
 
     return gpus
+
+
+def get_default_instance_type():
+    """If current region support free tier, return the free tier instance type. Otherwise, return t3.micro ."""
+    if not hasattr(get_default_instance_type, "cache"):
+        get_default_instance_type.cache = {}
+    cache = get_default_instance_type.cache
+    region = os.environ.get("AWS_DEFAULT_REGION")
+    if region not in cache:
+        ec2_client = boto3.client("ec2")
+        paginator = ec2_client.get_paginator("describe_instance_types")
+        page_iterator = paginator.paginate(
+            Filters=[
+                {"Name": "free-tier-eligible", "Values": ["true"]},
+                {"Name": "current-generation", "Values": ["true"]},
+            ]
+        )
+        free_tier_instance_type = []
+        for page in page_iterator:
```

Comment:
Why are we iterating all the results if we need just the first one?

Reply:
Yes. Even if we are just expecting a single result, boto3 will sometimes leave the first page blank. If we call boto3 directly, we will get an empty result.

Reply:
Got it. I would suggest the utility function [`paginate_boto_3`](https://github.com/aws/aws-parallelcluster/blob/3a66c3e2250fca6138f5ffd420bbdc298083f003/cli/src/pcluster/utils.py#L120) to simplify the code.